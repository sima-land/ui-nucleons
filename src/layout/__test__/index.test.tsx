import { render } from '@testing-library/react';
import { Layout, DesktopLayout, MobileLayout } from '..';

describe('Layout', () => {
  it('should renders correctly', () => {
    const { container } = render(
      <Layout className='test-class' style={{ height: 128 }}>
        Hello, world!
      </Layout>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should handle "disabledOn" prop', () => {
    const { container } = render(<Layout disabledOn={['xl', 'xs']}>Hello, world!</Layout>);

    expect(container).toMatchSnapshot();
  });
});

describe('Layouts (legacy)', () => {
  [DesktopLayout, MobileLayout].forEach(LayoutComponent => {
    it('should renders correctly', () => {
      const { container } = render(
        <LayoutComponent className='test-class' style={{ height: 128 }}>
          Hello, world!
        </LayoutComponent>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should handle "element" prop', () => {
      const { container } = render(
        <LayoutComponent element='main' className='test-class' style={{ height: 128 }}>
          Hello, world!
        </LayoutComponent>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should handle "disabledOn" prop', () => {
      const { container } = render(
        <LayoutComponent disabledOn={['xl', 'xs']}>Hello, world!</LayoutComponent>,
      );

      expect(container).toMatchSnapshot();
    });
  });
});
