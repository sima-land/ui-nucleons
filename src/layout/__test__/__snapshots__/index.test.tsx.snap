// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should handle "disabledOn" prop 1`] = `
<div>
  <div
    class="layout disabled-xl disabled-xs"
  >
    Hello, world!
  </div>
</div>
`;

exports[`Layout should renders correctly 1`] = `
<div>
  <div
    class="layout test-class"
    style="height: 128px;"
  >
    Hello, world!
  </div>
</div>
`;

exports[`Layouts (legacy) should handle "disabledOn" prop 1`] = `
<div>
  <div
    class="layout desktop disabled-xl disabled-xs"
  >
    Hello, world!
  </div>
</div>
`;

exports[`Layouts (legacy) should handle "disabledOn" prop 2`] = `
<div>
  <div
    class="layout mobile disabled-xl disabled-xs"
  >
    Hello, world!
  </div>
</div>
`;

exports[`Layouts (legacy) should handle "element" prop 1`] = `
<div>
  <main
    class="layout desktop test-class "
    style="height: 128px;"
  >
    Hello, world!
  </main>
</div>
`;

exports[`Layouts (legacy) should handle "element" prop 2`] = `
<div>
  <main
    class="layout mobile test-class "
    style="height: 128px;"
  >
    Hello, world!
  </main>
</div>
`;

exports[`Layouts (legacy) should renders correctly 1`] = `
<div>
  <div
    class="layout desktop test-class "
    style="height: 128px;"
  >
    Hello, world!
  </div>
</div>
`;

exports[`Layouts (legacy) should renders correctly 2`] = `
<div>
  <div
    class="layout mobile test-class "
    style="height: 128px;"
  >
    Hello, world!
  </div>
</div>
`;
